wallet/wallet.c

```diff
@@ -377,6 +378,43 @@ const struct utxo **wallet_select_coins(const tal_t *ctx, struct wallet *w,
 	return utxo;
 }
 
+const struct utxo **wallet_select_specific(const tal_t *ctx, struct wallet *w,
+					const char **outpoints)
+{
+	size_t i, j;
+	char **outpoint;
+	unsigned int output_index;
+	struct bitcoin_txid *txid;
+	struct utxo **available = wallet_get_utxos(ctx, w, output_state_available);
+	const struct utxo **utxos = tal_arr(ctx, const struct utxo*, 0);
+	tal_add_destructor2(utxos, destroy_utxos, w);
+
+	for (i = 0; i < tal_count(outpoints); i++)
+	{
+		outpoint = tal_strsplit(ctx, outpoints[i], ":", STR_NO_EMPTY);
+		txid = tal(ctx, struct bitcoin_txid);
+		if (!bitcoin_txid_from_hex(outpoint[0], strlen(outpoint[0]), txid)
+				|| opt_set_uintval(outpoint[1], &output_index))
+			return NULL;
+
+		for (j = 0; j < tal_count(available); j++) {
+			struct utxo *u = tal_steal(utxos, available[j]);
+
+			if (bitcoin_txid_eq(&u->txid, txid) && u->outnum == output_index) {
+				    tal_arr_expand(&utxos, u);
+			    if (!wallet_update_output_status(
+				    w, &available[j]->txid, available[j]->outnum,
+				    output_state_available, output_state_reserved))
+				    fatal("Unable to reserve output");
```

Comment:
Um, does this mean I crash the daemon if I try to fund two channels at once from the same UTXO?

Reply:
I think so, as in the already used function to select outputs : https://github.com/ElementsProject/lightning/blob/35827111ee837fc0118e0190b9e12539adae81b3/wallet/wallet.c#L315-L318
Anyway we consume the whole utxos if they are directly passed to `fundchannel` so one could not fund 2 channels, or am I missing something ?

Reply:
I agree with @darosior, line 388 of the new code is `wallet_get_utxos` which ensures that we can only change update status of UTXOs with `output_state_available`.  Will mark as resolved.